optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -205,16 +206,18 @@ private void processDifficulty(DescriptorPolicy descriptorPolicy, PlanningEntity
         }
     }
 
-    private void processValueRangeProviderAnnotation(DescriptorPolicy descriptorPolicy, Member member) {
+    private void processValueRangeProviderAnnotation(DomainAccessType domainAccessType, DescriptorPolicy descriptorPolicy,
+            Member member) {
         if (((AnnotatedElement) member).isAnnotationPresent(ValueRangeProvider.class)) {
-            MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(
+            MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(domainAccessType,
                     member, FIELD_OR_READ_METHOD, ValueRangeProvider.class);
             descriptorPolicy.addFromEntityValueRangeProvider(
                     memberAccessor);
         }
     }
 
-    private void processPlanningVariableAnnotation(DescriptorPolicy descriptorPolicy, Member member) {
+    private void processPlanningVariableAnnotation(DomainAccessType domainAccessType, DescriptorPolicy descriptorPolicy,
+            Member member) {
```

Comment:
Instead of passing the access type everywhere can't we just make it part of DescriptorPolicy?

The Policy objects are exactly created for this purpose: a bunch of general information that is passed to all the build/create/policy methods.